### ISSUE TITLE
405 Method Not Allowed instead of Not found when no root route is defined

### ISSUE THREAD

**Issue by Madeorsk** [mookums/zzz#17]
If we serve nothing on the root route, we would expect `serve_not_found` handler to be called, like any other route. Currently, we get a `405 Method Not Allowed` error instead.

Minimal reproduction by just taking the [basic example](https://github.com/mookums/zzz/blob/3fc7e5880c40d2fdb3925e3d4325e77a9dc82172/examples/basic/main.zig) and removing the "/" route.

```zig
const std = @import("std");

const zzz = @import("zzz");
const http = zzz.HTTP;

const tardy = @import("tardy");
const Tardy = tardy.Tardy(.auto);
const Runtime = tardy.Runtime;

const Server = http.Server(.plain);
const Router = Server.Router;
const Context = Server.Context;
const Route = Server.Route;

pub fn main() !void {
    const host: []const u8 = "0.0.0.0";
    const port: u16 = 9862;

    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = gpa.allocator();
    defer _ = gpa.deinit();

    // Creating our Tardy instance that
    // will spawn our runtimes.
    var t = try Tardy.init(.{
        .allocator = allocator,
        .threading = .single,
    });
    defer t.deinit();

    var router = Router.init(allocator);
    defer router.deinit();

    router.serve_not_found(Route.init().get({}, struct {
        fn handler_fn(ctx: *Context, _: void) !void {
            try ctx.respond(.{
                .status = .@"Not Found",
                .mime = http.Mime.HTML,
                .body = "Not Found Handler!",
            });
        }
    }.handler_fn));

    // This provides the entry function into the Tardy runtime. This will run
    // exactly once inside of each runtime (each thread gets a single runtime).
    try t.entry(
        &router,
        struct {
            fn entry(rt: *Runtime, r: *const Router) !void {
                var server = Server.init(rt.allocator, .{});
                try server.bind(.{ .ip = .{ .host = host, .port = port } });
                try server.serve(r, rt);
            }
        }.entry,
        {},
        struct {
            fn exit(rt: *Runtime, _: void) !void {
                try Server.clean(rt);
            }
        }.exit,
    );
}

```

**Comment by mookums** [comment]
Hello!

Thanks for reporting this issue. It was a pretty simple change regarding how the root node is treated within the RoutingTrie. It had a placeholder route attached that it would attempt to match. It should be fixed here (67d0d5929b7f171108dfceb8a81303f96f4d610d).

Feel free to try it out on `main`.

**Comment by Madeorsk** [comment]
Yes, quick and easy fix. Didn't notice the  `Route.init()` in `RoutingTrie`! Thanks!
